client/app/core/authorization.run.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-useless-return */
+/** @ngInject */
+
+export function authInit ($rootScope, $state, $log, Session, $sessionStorage, Language, ApplianceInfo, $window, RBAC, Redux, $ngRedux) {
+  $rootScope.$on('$stateChangeStart', changeStart)
+  $rootScope.$on('$stateChangeError', changeError)
+  $rootScope.$on('$stateChangeSuccess', changeSuccess)
+
+  $sessionStorage.$sync()  // needed when called right on reload
+  if ($sessionStorage.token) {
+    syncSession()
+  }
+
+  function changeStart (event, toState, toParams, _fromState, _fromParams) {
+    Redux.loadRedux().then(() => {
+      // first lets see if they are logged in and session is valid
+      event.preventDefault()
+      let nextState = ''
+      if (!Session.active()) {
```

Comment:
I'm not following this logic at all - can you expand on why its needed? is changeStart is an angular router state or redux state?

Reply:
Change start is an angular UI router event hook.  This whole section of code is refactoring of existing code.  I know the authorization.run.js file is new as of this PR so it seems like new code but it's not.  It is correcting an issue where we had combined an angular config file with an angular run block.  Some of the changes I did implement inside of it were for readability sake since this code hadn't been touched in a long time and needed some refactoring